openpype/hosts/blender/api/ops.py

```diff
@@ -432,9 +436,21 @@ def execute(self, context):
                 )
                 return {"CANCELLED"}
 
-            last_workfile_path, last_published_time = download_last_workfile()
-            if last_workfile_path:
-                bpy.ops.wm.open_mainfile(filepath=last_workfile_path)
+            # Get current project, asset and task names
+            project_name = get_current_project_name()
+            asset_name = get_current_asset_name()
+            task_name = get_current_task_name()
+
+            # Download and copy locally last workfile
+            last_workfile_path, last_published_time = download_last_workfile(
+                project_name, asset_name, task_name
+            )
+            local_workfile_path = save_as_local_workfile(
+                project_name, asset_name, task_name, last_workfile_path
+            )
+
+            if local_workfile_path:
+                bpy.ops.wm.open_mainfile(filepath=local_workfile_path)
```

Comment:
```suggestion
            if local_workfile_path := save_as_local_workfile(
                project_name, asset_name, task_name, last_workfile_path
            ):
```
Nitpicking but I think it's pretty cool that way :sunglasses: